crypto/src/main/java/net/corda/v5/crypto/ParameterizedSignatureSpec.java

```diff
@@ -14,12 +15,23 @@
  * When used for signing the [signatureName] must match the corresponding key scheme, e.g. you cannot use
  * "SHA256withECDSA" with "RSA" keys.
  */
-class ParameterizedSignatureSpec(
-    signatureName: String,
-    val params: AlgorithmParameterSpec
-) : SignatureSpec(signatureName) {
+public final class ParameterizedSignatureSpec extends SignatureSpec {
+    @NotNull private final AlgorithmParameterSpec params;
+
     /**
      * Converts a [ParameterizedSignatureSpec] object to a string representation.
      */
-    override fun toString(): String = "$signatureName:${params::class.java.simpleName}"
-}
\ No newline at end of file
+    @NotNull public String toString() {
+        return this.getSignatureName() + ':' + this.params.getClass().getSimpleName();
+    }
+
+    @NotNull
+    public final AlgorithmParameterSpec getParams() {
+        return this.params;
+    }
+
+    public ParameterizedSignatureSpec(@NotNull String signatureName, @NotNull AlgorithmParameterSpec params) {
+        super(signatureName);
+        this.params = params;
+    }
```

Comment:
Same, constructor is usually the first function.

Reply:
fixed